sensitivity_games/linear_feedback_controller.py

```diff
@@ -22,19 +22,21 @@ def solve_dare_for_k(self, Q, R):
         # http://www.mwm.im/lqr-controllers-with-python/
         # first, try to solve the ricatti equation
         X = torch.tensor(linalg.solve_discrete_are(self.dynamics.A,
-                                                   self.dynamics.B,
+                                                   self.dynamics.B.detach(),
                                                    Q,
                                                    R), dtype=torch.float32)
+
         # compute the LQR gain
-        term1 = torch.tensor(linalg.inv(self.dynamics.B.T@X@self.dynamics.B+R),
-                             dtype=torch.float32)
+        inv = linalg.inv((self.dynamics.B.T@X@self.dynamics.B).detach()+R)
```

Comment:
nit: add whitespace around arithmetic operations, i.e. `B.T @ X @ B`

Also, it's faster and more numerically stable to do something like `torch.linalg.solve(B.T @ X @ B, term2)` to get `K`.

Reply:
great. i'll change a bit. i'm new to working with `flake8 `, which caused me to break apart the calculation.